src/error/EntityNotFoundError.ts

```diff
@@ -7,9 +7,15 @@ import { InstanceChecker } from "../util/InstanceChecker"
  * Thrown when no result could be found in methods which are not allowed to return undefined or an empty set.
  */
 export class EntityNotFoundError extends TypeORMError {
+    public readonly entityClass: EntityTarget<any>
+    public readonly criteria: any
+
     constructor(entityClass: EntityTarget<any>, criteria: any) {
```

Comment:
That changes the contract of the class. The `any` exists today. I'd rather not change it and risk other issues.

Reply:
I may not cause that much of in issue I guess since EntityTarget is a generic. but maybe there could be a change like.
```ts
export class EntityNotFoundError<Entity = any> extends TypeORMError {
```
but in this example Entity is just a name of the type that will be put in the places where the name is.
![image](https://github.com/typeorm/typeorm/assets/4602609/c6756ee7-2d40-498f-8aec-2ab4528bea18)
this would happen without setting a default though.